### PR TITLE
SCRUM-4509 Validation of HTPExpressionDataSampleAnnotation

### PR DESCRIPTION
**Pull request by VarunReddy1111**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/dao/HTPExpressionDatasetSampleAnnotationDAO.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.alliancegenome.curation_api.dao;
+
+import org.alliancegenome.curation_api.dao.base.BaseSQLDAO;
+import org.alliancegenome.curation_api.model.entities.HTPExpressionDatasetSampleAnnotation;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class HTPExpressionDatasetSampleAnnotationDAO extends BaseSQLDAO<HTPExpressionDatasetSampleAnnotation> {
+
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/dao/HTPExpressionDatasetSampleAnnotationDAO.java:10]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Indent must use tab characters

File: src/main/java/org/alliancegenome/curation_api/dao/HTPExpressionDatasetSampleAnnotationDAO.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import org.alliancegenome.curation_api.dao.base.BaseSQLDAO;
+import org.alliancegenome.curation_api.model.entities.HTPExpressionDatasetSampleAnnotation;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class HTPExpressionDatasetSampleAnnotationDAO extends BaseSQLDAO<HTPExpressionDatasetSampleAnnotation> {
+    	
+    protected HTPExpressionDatasetSampleAnnotationDAO() {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/dao/HTPExpressionDatasetSampleAnnotationDAO.java:11]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Indent must use tab characters

File: src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/BioSampleGenomicInformationDTO.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.alliancegenome.curation_api.model.ingest.dto.fms;
+import org.alliancegenome.curation_api.model.ingest.dto.base.BaseDTO;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class BioSampleGenomicInformationDTO extends BaseDTO{
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/BioSampleGenomicInformationDTO.java:9]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' is not preceded with whitespace.

File: src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/MicroarraySampleDetailsDTO.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package org.alliancegenome.curation_api.model.ingest.dto.fms;
+
+import org.alliancegenome.curation_api.model.ingest.dto.base.BaseDTO;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class MicroarraySampleDetailsDTO extends BaseDTO{
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/MicroarraySampleDetailsDTO.java:10]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' is not preceded with whitespace.

File: src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTO.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.List;
+
+import org.alliancegenome.curation_api.model.ingest.dto.base.BaseDTO;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class HTPExpressionDatasetSampleAnnotationFmsDTO extends BaseDTO{
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTO.java:12]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' is not preceded with whitespace.

File: src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/HTPExpressionDatasetSampleAnnotationIngestFmsDTO.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.List;
+
+import org.alliancegenome.curation_api.model.ingest.dto.base.BaseDTO;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class HTPExpressionDatasetSampleAnnotationIngestFmsDTO extends BaseDTO{
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/HTPExpressionDatasetSampleAnnotationIngestFmsDTO.java:12]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' is not preceded with whitespace.

File: src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/HTPExpressionDatasetSampleAnnotationIngestFmsDTO.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    private MetaDataFmsDTO metaData;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/HTPExpressionDatasetSampleAnnotationIngestFmsDTO.java:13]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Indent must use tab characters

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/BulkLoadJobExecutor.java [hunk 8]
```diff
@@ -138,6 +140,8 @@ public void process(BulkLoadFileHistory bulkLoadFileHistory, Boolean cleanUp) th
 			expressionAtlasExecutor.execLoad(bulkLoadFileHistory);
 		} else if (bulkLoadFileHistory.getBulkLoad().getBackendBulkLoadType() == BackendBulkLoadType.BIOGRID_ORCS) {
 			biogridOrcExecutor.execLoad(bulkLoadFileHistory);
+		} else if(bulkLoadFileHistory.getBulkLoad().getBackendBulkLoadType() == BackendBulkLoadType.HTPDATASAMPLE) {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/BulkLoadJobExecutor.java:143]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+import org.alliancegenome.curation_api.services.HTPExpressionDatasetSampleAnnotationService;
+import org.alliancegenome.curation_api.util.ProcessDisplayHelper;
+import org.apache.commons.lang3.StringUtils;
+
+import io.quarkus.logging.Log;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+@ApplicationScoped
+public class HTPExpressionDatasetSampleAnnotationExecutor extends LoadFileExecutor{
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java:29]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' is not preceded with whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+	@Inject ExternalDataBaseEntityService externalDataBaseEntityService;
+	@Inject ExternalDataBaseEntityDAO externalDataBaseEntityDAO;
+	@Inject HTPExpressionDatasetSampleAnnotationService htpExpressionDatasetSampleAnnotationService;
+	@Inject HTPExpressionDatasetSampleAnnotationDAO htpExpressionDatasetSampleAnnotationDAO;
+
+    public void execLoad(BulkLoadFileHistory bulkLoadFileHistory) {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java:36]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Indent must use tab characters

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        try {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java:37]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Indent must use tab characters

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+			bulkLoadFileHistory.finishLoad();
+
+			updateHistory(bulkLoadFileHistory);
+			updateExceptions(bulkLoadFileHistory);
+
+		} catch (Exception e) {
+			failLoad(bulkLoadFileHistory, e);
+			e.printStackTrace();
+		}
+    }
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java:72]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Indent must use tab characters

**Comment by VarunReddy1111** [comment]
Having conflict in migration file version with other PR, will change the version of this PR, if the other PR goes in first. If this PR goes first, will update the migration version of the  other PR after checking with Christian.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+			updateHistory(bulkLoadFileHistory);
+			updateExceptions(bulkLoadFileHistory);
+
+		} catch (Exception e) {
+			failLoad(bulkLoadFileHistory, e);
+			e.printStackTrace();
+		}
+	}
+
+	private boolean runLoad(BulkLoadFileHistory history, BackendBulkDataProvider dataProvider, List<HTPExpressionDatasetSampleAnnotationFmsDTO> htpDatasetSampleAnnotations, List<Long> htpAnnotationsIdsLoaded) {
```
Comment (markquintontulloch) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java]: I think we should be able to use the generic runLoad method here?

File: src/main/java/org/alliancegenome/curation_api/model/entities/HTPExpressionDatasetSampleAnnotation.java [hunk 14]
```diff
@@ -115,7 +115,7 @@ public class HTPExpressionDatasetSampleAnnotation extends AuditedObject {
 	private VocabularyTerm sequencingFormat;
 
 	@IndexingDependency(reindexOnUpdate = ReindexOnUpdate.SHALLOW)
-	@ManyToMany
+	@ManyToMany(cascade = CascadeType.ALL)
```
Comment (markquintontulloch) [inline src/main/java/org/alliancegenome/curation_api/model/entities/HTPExpressionDatasetSampleAnnotation.java:118]: Probably also want `orphanRemove = true` here

File: src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/BioSampleAgeDTO.java [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.alliancegenome.curation_api.model.ingest.dto.base.BaseDTO;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class BioSampleAgeDTO extends BaseDTO {
+	private WhenExpressedDTO stage;
```
Comment (markquintontulloch) [inline src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/BioSampleAgeDTO.java]: I know this isn't your code, but could we rename the WhenExpressedDTO and WhereExpressedDTO classes to WhenExpressedFmsDTO and WhereExpressedFmsDTO respectively.  That makes it clear that these are not expected to align with the LinkML schema.

File: src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/BioSampleGenomicInformationDTO.java [hunk 16]
```diff
@@ -0,0 +1,13 @@
+package org.alliancegenome.curation_api.model.ingest.dto.fms;
+import org.alliancegenome.curation_api.model.ingest.dto.base.BaseDTO;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class BioSampleGenomicInformationDTO extends BaseDTO {
```
Comment (markquintontulloch) [inline src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/BioSampleGenomicInformationDTO.java]: Should be rename to BioSampleGenomicInformationFmsDTO

File: src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/MicroarraySampleDetailsDTO.java [hunk 17]
```diff
@@ -0,0 +1,13 @@
+package org.alliancegenome.curation_api.model.ingest.dto.fms;
+
+import org.alliancegenome.curation_api.model.ingest.dto.base.BaseDTO;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class MicroarraySampleDetailsDTO extends BaseDTO {
```
Comment (markquintontulloch) [inline src/main/java/org/alliancegenome/curation_api/model/ingest/dto/fms/MicroarraySampleDetailsDTO.java]: Should be renamed to MicroarraySampleDetailsFmsDTO

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java [hunk 18]
```diff
@@ -0,0 +1,257 @@
...
+	@Inject AlleleService alleleService;
+	@Inject AffectedGenomicModelService affectedGenomicModelService;
+	@Inject NcbiTaxonTermService ncbiTaxonTermService;
+
+	@Transactional
+	public HTPExpressionDatasetSampleAnnotation validateHTPExpressionDatasetSampleAnnotationFmsDTO(HTPExpressionDatasetSampleAnnotationFmsDTO dto, BackendBulkDataProvider backendBulkDataProvider) throws ValidationException {
+		ObjectResponse<HTPExpressionDatasetSampleAnnotation> htpSampleAnnotationResponse = new ObjectResponse<>();
+		HTPExpressionDatasetSampleAnnotation htpSampleAnnotation;
+
+		if (dto.getSampleId().getPrimaryId() != null) {
```
Comment (markquintontulloch) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java]: I prefer to use StringUtils.isNotBlank for checks on strings as it rules out empty or whitespace-only strings as well as nulls.

**Review by markquintontulloch — CHANGES_REQUESTED** [review]
Looks good, just a few small changes requested

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+
+import org.alliancegenome.curation_api.dao.ExternalDataBaseEntityDAO;
+import org.alliancegenome.curation_api.dao.HTPExpressionDatasetSampleAnnotationDAO;
+import org.alliancegenome.curation_api.enums.BackendBulkDataProvider;
+import org.alliancegenome.curation_api.exceptions.ObjectUpdateException;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java:11]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - org.alliancegenome.curation_api.exceptions.ObjectUpdateException.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+import org.alliancegenome.curation_api.exceptions.ObjectUpdateException.ObjectUpdateExceptionData;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java:12]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - org.alliancegenome.curation_api.exceptions.ObjectUpdateException.ObjectUpdateExceptionData.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+import org.alliancegenome.curation_api.interfaces.AGRCurationSchemaVersion;
+import org.alliancegenome.curation_api.model.entities.HTPExpressionDatasetSampleAnnotation;
+import org.alliancegenome.curation_api.model.entities.bulkloads.BulkFMSLoad;
+import org.alliancegenome.curation_api.model.entities.bulkloads.BulkLoadFileHistory;
+import org.alliancegenome.curation_api.model.ingest.dto.fms.HTPExpressionDatasetSampleAnnotationFmsDTO;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java:17]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - org.alliancegenome.curation_api.model.ingest.dto.fms.HTPExpressionDatasetSampleAnnotationFmsDTO.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+import org.alliancegenome.curation_api.model.ingest.dto.fms.HTPExpressionDatasetSampleAnnotationIngestFmsDTO;
+import org.alliancegenome.curation_api.services.ExternalDataBaseEntityService;
+import org.alliancegenome.curation_api.services.HTPExpressionDatasetSampleAnnotationService;
+import org.alliancegenome.curation_api.util.ProcessDisplayHelper;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java:21]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - org.alliancegenome.curation_api.util.ProcessDisplayHelper.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+import org.apache.commons.lang3.StringUtils;
+
+import io.quarkus.logging.Log;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetSampleAnnotationExecutor.java:24]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - io.quarkus.logging.Log.

**Review by markquintontulloch — APPROVED** [review]

**Review by oblodgett — APPROVED** [review]

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java [hunk 24]
```diff
@@ -0,0 +1,282 @@
...
+				TemporalContext temporalContext = geneExpressionAnnotationFmsDTOValidator.updateTemporalContext(temporalContextObjectResponse, htpSampleAnnotation.getHtpExpressionSampleAge().getStage());
+				htpSampleAnnotation.getHtpExpressionSampleAge().setAge(dto.getSampleAge().getAge());
+				htpSampleAnnotation.getHtpExpressionSampleAge().setWhenExpressedStageName(dto.getSampleAge().getStage().getStageName());
+				htpSampleAnnotation.getHtpExpressionSampleAge().setStage(temporalContext);
+			}
+		}
+
+		List<Long> idsToRemove = new ArrayList<>();
+		if (CollectionUtils.isNotEmpty(dto.getSampleLocations())) {
+			if(CollectionUtils.isNotEmpty(htpSampleAnnotation.getHtpExpressionSampleLocations())) {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:129]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java [hunk 25]
```diff
@@ -0,0 +1,292 @@
...
+	@Inject AffectedGenomicModelService affectedGenomicModelService;
+	@Inject NcbiTaxonTermService ncbiTaxonTermService;
+	@Inject AnatomicalSiteDAO anatomicalSiteDAO;
+
+	@Transactional
+	public HTPExpressionDatasetSampleAnnotation validateHTPExpressionDatasetSampleAnnotationFmsDTO(HTPExpressionDatasetSampleAnnotationFmsDTO dto, BackendBulkDataProvider backendBulkDataProvider) throws ValidationException {
+		ObjectResponse<HTPExpressionDatasetSampleAnnotation> htpSampleAnnotationResponse = new ObjectResponse<>();
+		HTPExpressionDatasetSampleAnnotation htpSampleAnnotation;
+
+		Boolean sampleExists = ((dto.getSampleId() != null && StringUtils.isNotEmpty(dto.getSampleId().getPrimaryId())) || StringUtils.isNotEmpty(dto.getSampleTitle()));
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:69]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.UnnecessaryParenthesesCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary parentheses around assignment right-hand side.

Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:69]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.UnnecessaryParenthesesCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary parentheses around expression.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java [hunk 26]
```diff
@@ -0,0 +1,292 @@
...
+
+		if(!sampleExists) {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:71]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java [hunk 27]
```diff
@@ -0,0 +1,292 @@
...
+					htpSampleAnnotationResponse.addErrorMessage("SampleLocations", anatomicalSiteObjectResponse.errorMessagesString());
+				} else {
+					htpSampleLocations.add(anatomicalSiteObjectResponse.getEntity());
+				}
+			}
+			htpSampleAnnotation.setHtpExpressionSampleLocations(htpSampleLocations);
+		}
+
+		if (dto.getGenomicInformation() != null) {
+			Boolean genomicInformationExists = (StringUtils.isNotEmpty(dto.getGenomicInformation().getBioSampleText()) == true || StringUtils.isNotEmpty(dto.getGenomicInformation().getBiosampleId()) == true);
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:152]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.UnnecessaryParenthesesCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary parentheses around assignment right-hand side.

Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:152]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.SimplifyBooleanExpressionCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expression can be simplified.

Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:152]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.SimplifyBooleanExpressionCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expression can be simplified.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java [hunk 28]
```diff
@@ -0,0 +1,292 @@
...
+	@Inject AffectedGenomicModelService affectedGenomicModelService;
+	@Inject NcbiTaxonTermService ncbiTaxonTermService;
+	@Inject AnatomicalSiteDAO anatomicalSiteDAO;
+
+	@Transactional
+	public HTPExpressionDatasetSampleAnnotation validateHTPExpressionDatasetSampleAnnotationFmsDTO(HTPExpressionDatasetSampleAnnotationFmsDTO dto, BackendBulkDataProvider backendBulkDataProvider) throws ValidationException {
+		ObjectResponse<HTPExpressionDatasetSampleAnnotation> htpSampleAnnotationResponse = new ObjectResponse<>();
+		HTPExpressionDatasetSampleAnnotation htpSampleAnnotation;
+
+		Boolean sampleExists = (dto.getSampleId() != null && StringUtils.isNotEmpty(dto.getSampleId().getPrimaryId()) || StringUtils.isNotEmpty(dto.getSampleTitle()));
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:69]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.UnnecessaryParenthesesCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary parentheses around assignment right-hand side.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java [hunk 29]
```diff
@@ -0,0 +1,292 @@
...
+					htpSampleAnnotationResponse.addErrorMessage("SampleLocations", anatomicalSiteObjectResponse.errorMessagesString());
+				} else {
+					htpSampleLocations.add(anatomicalSiteObjectResponse.getEntity());
+				}
+			}
+			htpSampleAnnotation.setHtpExpressionSampleLocations(htpSampleLocations);
+		}
+
+		if (dto.getGenomicInformation() != null) {
+			Boolean genomicInformationExists = (StringUtils.isNotEmpty(dto.getGenomicInformation().getBioSampleText()) == true) || (StringUtils.isNotEmpty(dto.getGenomicInformation().getBiosampleId()) == true);
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:152]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.UnnecessaryParenthesesCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary parentheses around expression.

Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:152]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.SimplifyBooleanExpressionCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expression can be simplified.

Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:152]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.UnnecessaryParenthesesCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary parentheses around expression.

Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:152]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.SimplifyBooleanExpressionCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expression can be simplified.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java [hunk 30]
```diff
@@ -0,0 +1,292 @@
...
+	@Inject AffectedGenomicModelService affectedGenomicModelService;
+	@Inject NcbiTaxonTermService ncbiTaxonTermService;
+	@Inject AnatomicalSiteDAO anatomicalSiteDAO;
+
+	@Transactional
+	public HTPExpressionDatasetSampleAnnotation validateHTPExpressionDatasetSampleAnnotationFmsDTO(HTPExpressionDatasetSampleAnnotationFmsDTO dto, BackendBulkDataProvider backendBulkDataProvider) throws ValidationException {
+		ObjectResponse<HTPExpressionDatasetSampleAnnotation> htpSampleAnnotationResponse = new ObjectResponse<>();
+		HTPExpressionDatasetSampleAnnotation htpSampleAnnotation;
+
+		Boolean sampleExists = (dto.getSampleId() != null && StringUtils.isNotEmpty(dto.getSampleId().getPrimaryId())) || StringUtils.isNotEmpty(dto.getSampleTitle());
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java:69]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.UnnecessaryParenthesesCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary parentheses around expression.

**Review by abecerra — APPROVED** [review]

**Pull request merged**